main.tf

```diff
@@ -1,50 +1,56 @@
 module "s3" {
   source     = "./modules/s3"
-  model_name = var.model_name
+  model_name = local.model_name
   tags       = var.tags
 }
 
 
 module "sagemaker" {
   source                       = "./modules/sagemaker"
   tags                         = var.tags
-  sagemaker_execution_role_arn = module.iam.sagemaker_role_arn
+  sagemaker_execution_role_arn = module.iam.sagemaker_role.arn
   model_target_variable        = var.model_target_variable
   algorithm_choice             = var.algorithm_choice
   tuning_metric                = var.tuning_metric
-  config_bucket_id             = module.s3.config_bucket_id
+  config_bucket_id             = module.s3.config_bucket.id
+  data_s3_bucket               = var.data_s3_bucket
   data_location_s3             = var.data_location_s3
-  endpoint_name                = var.endpoint_name
-  model_name                   = var.model_name
+  endpoint_name                = local.endpoint_name
+  model_name                   = local.model_name
   sagemaker_instance_type      = var.sagemaker_instance_type
   model_instance_count         = var.model_instance_count
-  ecr_repo_uri                 = module.ecr.ecr_repo_uri
+  pycaret_image_uri            = "${module.ecr.repository.repository_url}:latest"
 }
 
 
 module "iam" {
-  source     = "./modules/iam"
-  tags       = var.tags
-  region     = var.region
-  account_id = var.account_id
-  model_name = var.model_name
+  source                = "./modules/iam"
+  tags                  = var.tags
+  model_name            = local.model_name
+  config_s3_bucket      = module.s3.config_bucket.id
```

Comment:
Passing in each of the buckets and associated keys allows us to construct a fine grained IAM policy, as opposed to giving our roles access to all S3 buckets and KMS keys